libsolidity/ast/Types.cpp

```diff
@@ -390,10 +561,11 @@ IntegerConstantType::IntegerConstantType(Literal const& _literal)
 	}
 }
 
-bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+bool RationalNumberType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	if (auto targetType = dynamic_cast<IntegerType const*>(&_convertTo))
+	if (_convertTo.category() == Category::Integer)
 	{
+		auto targetType = dynamic_cast<IntegerType const*>(&_convertTo);
```

Comment:
in this case though, there is an implicit conversion, we simply truncate off the fractional bits.